### PR TITLE
Add user to database with hashed password

### PR DESCRIPTION
**Pull request by wasanderes94**
- Add bcrypt library for hashing passwords & verifying hashes
- Insert user into db if sign up was successful

### CODE REVIEW

File: app/src/main/java/org/feature/fox/coffee_counter/ui/authentication/AuthenticationViewModel.kt [hunk 1]
```diff
@@ -76,8 +78,20 @@ class AuthenticationViewModel @Inject constructor(
...
+        } else {
+            userRepository.insertUser(
+                User(
+                    id = idState.value.text,
+                    name = nameState.value.text,
+                    false,
+                    password = BCrypt.withDefaults()
+                        .hashToString(12, passwordState.value.text.toCharArray())
+                )
+            )
```
Comment (Morphclue) [inline app/src/main/java/org/feature/fox/coffee_counter/ui/authentication/AuthenticationViewModel.kt:90]: No need for an else case if the `if` already has a return inside.

File: app/src/main/java/org/feature/fox/coffee_counter/ui/authentication/AuthenticationViewModel.kt [hunk 2]
```diff
@@ -76,8 +78,20 @@ class AuthenticationViewModel @Inject constructor(
...
         }
 
+
+
```
Comment (Morphclue) [inline app/src/main/java/org/feature/fox/coffee_counter/ui/authentication/AuthenticationViewModel.kt:94]: Maximum one empty line between code inside a function to seperate logical blocks is the as far as i know the current standard.

**Review by Morphclue — CHANGES_REQUESTED** [review]
Aside from the small annotations, the PR looks good!

File: app/src/main/java/org/feature/fox/coffee_counter/ui/authentication/AuthenticationViewModel.kt (hunk 1 above)
Comment (wasanderes94) [reply app/src/main/java/org/feature/fox/coffee_counter/ui/authentication/AuthenticationViewModel.kt]: Should be resolved with ec8b6109016cacdaa8ebd4489789fc55f978eb85

**Pull request merged**